include/reactor-uc/network_channel.h

```diff
@@ -95,6 +107,19 @@ struct NetworkChannel {
   void (*free)(NetworkChannel *self);
 };
 
+struct SyncNetworkChannel {
```

Comment:
The `NetworkChannelMode` enum is either `poll` or `async`, but the `NetworkChannel` subtypes are `SyncNetworkChannel` or `AsyncNetworkChannel`

Maybe it makes sense to align these and either have:
`NetworkChannelMode` = `Async` | `Sync`

or have:

`SyncNetworkChannel` and `PollNetworkChannel`